### PR TITLE
Adding a solr:create rake task

### PR DESCRIPTION
**Pull request by bertrama**
It looks like there aren't many good options for parsing arguments in rake.
This produces the cleanest command line invocation:
```bash
rake solr:create <directory> <name>
```

### CODE REVIEW

**Comment by bertrama** [comment]
If preferred, I can change the task to use rake's native argument handling.  i.e. invocation with:

```bash
prompt$ rake solr:create[directory, name]
```

**Comment by jcoyne** [comment]
I would like to see this called `solr:collection:create` as `solr:create` sounds like what you'd do to create a server.

File: lib/solr_wrapper/tasks/solr_wrapper.rake [hunk 1]
```diff
@@ -32,6 +32,21 @@ namespace :solr do
...
+    start = ARGV.find_index('solr:create')
+    if ARGV.length >= start + 2
+      directory = ARGV[start + 1]
+      name      = ARGV[start + 2]
+      # Rake tries to run these as tasks.
+      task(directory.to_sym) {}
+      task(name.to_sym) {}
+      @solr_instance.create dir: directory, name: name
+    else
+      raise "Usage: rake solr:create <directory> <name>"
```
Comment (jcoyne) [inline lib/solr_wrapper/tasks/solr_wrapper.rake]: This shouldn't be an exception, it should just log to `$stderr`

**Comment by jcoyne** [comment]
I don't understand why you'd want to do all this argument handling in rake when you have a binary script that is a lot more flexible at it.

**Comment by cbeer** [comment]
I wonder if we need to expose the path to the unpacked `bin/solr` command.

**Comment by bertrama** [comment]
To explain my motivation, I was following along the [Dive into Hydra](https://github.com/projecthydra/hydra/wiki/Lesson---Start-FCRepo-and-Solr) guide, but every time I stopped and restarted solr, it would lose the data in the collection.

I also noticed that `rake solr:start` and `rake solr:stop` don't have the same behavior as solr_wrapper in regard to deleting the collection when it stopped, so it made sense to me to do a matching rake task.

Using `bin/solr` makes sense until I think about running two demos.  Then have to remember and specify the port.

**Comment by jcoyne** [comment]
> every time I stopped and restarted solr, it would lose the data in the collection

@cbeer do you know how to make collections persistent in Solr 5/6?

**Comment by cbeer** [comment]
By design, it does not persist instance data. It might be possible to add configuration to disable that behavior, but we'd want to think hard about whether that's a good idea. In most cases where we're interested in persistence, I think we ought to encourage people to run solr directly.

**Comment by cam156** [comment]
@cbeer, @jcoyne: I know I did something very similar so I could could just run rake solr:start in development and not need to worry about losing the state of my solr instance all the time.

This PR comes close to allowing the user to run rake solr:start (like we used to run rake jetty:start) and you are up for development.

I was just about to put in a very similar PR when i realized this one already existed: https://github.com/cbeer/solr_wrapper/compare/master...psu-stewardship:solr_config?expand=1

**Comment by cam156** [comment]
@bertrama The new preference would be to use the executable.  @cbeer included a new flag -preserve that keeps data from one run to the next.  Can we close this PR?

In addition we changed the Active Fedora Generator to add this flag by default when the test app is generated in sufia.  There is some documentation on running with the new configuration files here: https://github.com/projecthydra/sufia/wiki/Sufia-Development-Guide#work-with-test-app-in-the-browser

**Comment by bertrama** [comment]
Yeah, that'll work for me, thanks!

**Pull request closed without merging**